metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,303 @@
+              required=False,
+              default=None,
+              type=str,
+              help="Change namespace from the default (your username) to "
+                   "the specified tag.")
+@click.argument('tags',
+                required=True,
+                type=str,
+                nargs=-1)
+@click.pass_obj
+def remove(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = obtain_metaflow_run_client(obj,
+                                                     run_id, user_namespace)
+
+    metaflow_run_client.remove_tag(list(tags))
+
+    obj.echo("Operation successful. New tags:")
+    obj.echo(print_tags(metaflow_run_client))
+
+@tag.command('replace',
+             help='Replace tags of a run.')
+@click.option('--run-id',
+              required=False,  # set False here so we can throw a better error message
+              default=None,
+              type=str,
+              help="Run ID of the specific run to tag.  [required]")
+@click.option('--namespace',
+              'user_namespace',
+              required=False,
+              default=None,
+              type=str,
+              help="Change namespace from the default (your username) to "
+                   "the specified tag.")
+@click.argument('tags',
+                required=True,
+                type=str,
+                nargs=2)
+@click.pass_obj
+def replace(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = obtain_metaflow_run_client(obj,
+                                                     run_id, user_namespace)
+
+    metaflow_run_client.replace_tag([tags[0]], [tags[1]])
+
+    obj.echo("Operation successful. New tags:")
+    obj.echo(print_tags(metaflow_run_client))
+
+@tag.command('list',
+             help='List tags of a run.')
+@click.option('--run-id',
+              required=False,
+              default=None,
+              type=str,
+              help="Run ID of the specific run to list.")
+@click.option('--all',
+              'list_all',
+              required=False,
+              is_flag=True,
+              default=False,
+              help="List tags across all runs of this flow.")
+@click.option('--my-runs',
+              'my_runs',
+              required=False,
+              is_flag=True,
+              default=False,
+              help="List tags across all runs of the flow under the default namespaces.")
+@click.option('--hide-system-tags',
+              required=False,
+              is_flag=True,
+              default=False,
+              help="Hide system tags.")
+@click.pass_obj
+def tag_list(obj, run_id, hide_system_tags, list_all, my_runs):
+    if (run_id is None and not list_all and not my_runs):
+        raise CommandException(
+            "Please use one of the options --run-id / --all / --my-runs.\n" +
```

Comment:
let's default to `--all` - `python tag list` should return something by default instead of showing an error